packages/views.py

```diff
@@ -0,0 +1,281 @@
+        actual_package.update(origin=package.origin, destination=package.destination, rate=package.rate,
+                              full_name=package.full_name, phone_number=package.phone_number, notes=package.notes)
+
+    return packages
+
+
+def create_saved_packages(request):
+    packages = serializers.deserialize('json', request.COOKIES.get('new_packages'))
+
+    for package in packages:
+        Package.objects.create(order=package.order, origin=package.origin, destination=package.destination,
+                               rate=package.rate, full_name=package.full_name, phone_number=package.phone_number,
+                               notes=package.notes)
+
+    return packages
+
+
+def send_emails(updated_packages, new_packages):
+    # TODO:
+    # send out emails where we need
+    pass
+
+
+def save_changes_to_cookies(request, order):
+    origin_address = get_or_create_origin_address(request)
+    destination_address = get_or_create_destination_address(request)
+
+    rate = float(request.POST.get('rate').replace('₪', ''))
+    phone_number = request.POST.get('phone_number')
+    full_name = request.POST.get('full_name')
+    notes = request.POST.get('notes')
+
+    package_id = request.POST.get('package_id')
+
+    package = Package(origin=origin_address, destination=destination_address,rate=rate, phone_number=phone_number,
+                      full_name=full_name, order=order, notes=notes)
+    if package_id:
+        package.id = package_id
+        json = request.COOKIES.get('updated_packages')
+        json = add_package_to_json(package, json)
+        request.set_cookie('updated_packages', json)
+    else:
+        json = request.COOKIES.get('new_packages')
+        json = add_package_to_json(package, json)
+        request.set_cookie('new_packages', json)
+
+
+def get_or_create_destination_address(request):
+    destination_city_name = request.POST.get('destination_city')
+    destination_area = request.POST.get('destination_area')
+    destination_street = request.POST.get('destination_street')
+    destination_street_number = request.POST.get('destination_street_number')
+
+    return get_or_create_address(destination_city_name, destination_area, destination_street, destination_street_number)
+
+
+def get_or_create_origin_address(request):
+    origin_city_name = request.POST.get('origin_city')
+    origin_area = request.POST.get('origin_area')
+    origin_street = request.POST.get('origin_street')
+    origin_street_number = request.POST.get('origin_street_number')
+
+    return get_or_create_address(origin_city_name, origin_area, origin_street, origin_street_number)
+
+
+def get_or_create_address(city_name, area, street_name, street_number):
+    destination_city = get_or_create_city(city_name, area)
+    return Address.objects.get_or_create(city=destination_city, street_name=street_name, street_number=street_number)[0]
+
+
+def get_or_create_city(name, area):

```

Comment:
You might find a problem where people misspell address and cities and then you have both
`בית שמש` and `בת שמש` which refers to the same location. 

It would be better to link this to a location API and fill location before-hand, so that all known cities are already populated in the database. 

This is okay to be left for later, just make an issue or comment about it somewhere.
(also true about addresses).

Reply:
I didn't think that misspelling of cities (or streets) will be an issue because these inputs use a data list to autocomplete the name. But I still agree, I'll create an issue for cities and streets.